src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -27,74 +52,603 @@ const ActorInit En_Dns_InitVars = {
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 0;
+    this->actor.gravity = -0.8f;
+    this->unk_2D2 = 0;
+    this->unk_2C6 = 0;
+    func_8013AED4(&this->unk_2C6, 3, 7);
+    this->unk_2C6 |= (0x40 | 0x10);
+    this->unk_2C6 |= 0x200;
+    if (gSaveContext.weekEventReg[9] & 0x80) {
+        this->unk_2FC = 1;
+    } else {
+        this->unk_2FC = 0;
+    }
+    this->actionFunc = func_8092D1B8;
+    gSaveContext.eventInf[1] &= (u8)~0x20;
+    gSaveContext.eventInf[1] &= (u8)~0x40;
+}
+
+void EnDns_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDns_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    if (!func_8092CAD0(this, globalCtx) && func_8092CB98(this, globalCtx)) {
+        func_8092D5E8(this, globalCtx);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        func_8092C5C0(this);
+        return;
+    }
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_8092C934(this);
+    func_8092C86C(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+    func_8013C964(&this->actor, globalCtx, 80.0f, 40.0f, 0, this->unk_2C6 & 7);
+    Actor_SetHeight(&this->actor, 34.0f);
+    func_8092C6FC(this, globalCtx);
+    func_8092C5C0(this);
+}
+
+s32 func_8092D954(s16 arg0, s16 arg1, Vec3f* arg2, Vec3s* arg3, s32 arg4, s32 arg5) {
+    Vec3f sp74;
+    Vec3s sp6C;
+    MtxF sp2C;
+
+    SysMatrix_MultiplyVector3fByState(&D_801D15B0, &sp74);
+    SysMatrix_CopyCurrentState(&sp2C);
+    func_8018219C(&sp2C, &sp6C, 0);
+    *arg2 = sp74;
+
+    if (arg4 == 0) {
+        if (arg5 != 0) {
+            sp6C.z = arg0;
+            sp6C.y = arg1;
+        }
+        Math_SmoothStepToS(&arg3->x, sp6C.x, 3, 0x2AA8, 0xB6);
+        Math_SmoothStepToS(&arg3->y, sp6C.y, 3, 0x2AA8, 0xB6);
+        Math_SmoothStepToS(&arg3->z, sp6C.z, 3, 0x2AA8, 0xB6);
+    } else {
+        arg3->x = sp6C.x;
+        arg3->y = sp6C.y;
+        arg3->z = sp6C.z;
+    }
+
+    return 1;
+}
+
+s32 func_8092DA68(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDns* this = THIS;
+
+    this->unk_1E4[limbIndex] = *dList;
+    *dList = NULL;
+    return 0;
+}
+
+void func_8092DA94(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDns* this = THIS;
+    s32 pad;
+    s32 phi_v1;
+    s32 phi_v0;
+
+    if (this->unk_2C6 & 0x10) {
+        phi_v1 = 1;
+        phi_v0 = 0;
+    } else {
+        phi_v1 = 0;
+        if (this->unk_2C6 & 0x20) {
+            phi_v0 = 1;
+        } else {
+            phi_v0 = 0;
+        }
+    }
+
+    if (limbIndex == 2) {
+        func_8092D954(this->unk_2CC, (s16)(this->unk_2CE + this->actor.shape.rot.y), &this->unk_218, &this->unk_224,
```

Comment:
```suggestion
        func_8092D954(this->unk_2CC, this->unk_2CE + this->actor.shape.rot.y, &this->unk_218, &this->unk_224,
```
The cast seems unnecessary since the function already takes s16.